server/src/main/java/io/unitycatalog/server/service/AuthService.java

```diff
@@ -167,17 +168,37 @@ public HttpResponse grantToken(
             String cookieTimeout =
                 ServerProperties.getInstance().getProperty("server.cookie-timeout", "P5D");
             Cookie cookie =
-                Cookie.secureBuilder(AuthDecorator.UC_TOKEN_KEY, accessToken)
-                    .path("/")
-                    .maxAge(Duration.parse(cookieTimeout).getSeconds())
-                    .build();
+                createCookie(
+                    AuthDecorator.UC_TOKEN_KEY,
+                    accessToken,
+                    "/",
+                    Duration.parse(cookieTimeout).getSeconds());
```

Comment:
Can this be moved to be part of `createCookie`?

Reply:
+1, the server property seems to be used only in createCookie.

Reply:
+1

Reply:
main idea behind put this in a method  as it is used  during login where we  set and other during logout where we  expire , the difference is during logout we need to set age to 0  and api accepts only Long  where as during login we accept duration in days .. so we need to convert it to long and then call the  create cookie method

Reply:
Why not put the `Duration.parse(...).getSeconds()` part into `createCookie` so that you just pass in the duration string into `createCookie`? And then when you want to set age to zero, there should be a similar Duration syntax for that? `PT0S` maybe?

Reply:
ahh got it , didnt strike that 0 can be set in the format :)